mautic/mautic
# SMS click tracking and contact identification

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? |
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6747 https://github.com/mautic/mautic/issues/6855
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This adds support to identify contacts from tracking links in text messages.

#### Steps to test this PR:
1. Configure Twilio
2. (You will likely need to us bit.ly for the url shortening service as the URL itself can be too long for a single SMS)
3. Create a text message with a link
4. Create a campaign to send the text message and add a contact with a mobile number
5. You should receive the text message with a redirect link
6. Open the URL
7. The click should be recorded in the  contact timeline
8. Test links in emails are still tracked against the contact since tracking was moved into an event subscriber
